src/main/java/io/odpf/firehose/sink/common/ProtoUtil.java

```diff
@@ -0,0 +1,33 @@
+package io.odpf.firehose.sink.common;
+
+import com.google.protobuf.DynamicMessage;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ProtoUtil {
+    public static boolean isUnknownFieldExist(DynamicMessage dynamicMessage) {
+        if (dynamicMessage == null) {
+            return false;
+        }
+        List<DynamicMessage> dynamicMessageFields = new LinkedList<>();
```

Comment:
missing the root message

Reply:
similar with beast changes, every node of DynamicMessage will be added automatically when traversing the DynamicMessage trees, so there is no need to add the root explicitly on accumulator

Reply:
missed it.